util/progress/progressui/json.go

```diff
@@ -0,0 +1,44 @@
+package progressui
+
+import (
+	"time"
+
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/solver/pb"
+)
+
+type Vertex struct {
+	*client.Vertex
+	Intervals []VertexInterval       `json:"intervals,omitempty"`
+	Statuses  []VertexStatus         `json:"statuses,omitempty"`
+	Logs      map[string][]VertexLog `json:"logs,omitempty"`
+	Warnings  []VertexWarning        `json:"warnings,omitempty"`
+}
+
+type VertexInterval struct {
+	Start time.Time `json:"start"`
```

Comment:
Do you mean use `*time.Time`? This structure is intended to mimic `interval` which is defined as:

```go
type interval struct {
    start *time.Time
    stop *time.Time
}
```

I modified it to use a non-pointer because this version of the type should not have nullable versions of start and stop.

Reply:
No, I meant that in current types these fields are called `Started` and `Completed`

Reply:
Had an offline conversation regarding this. The inclusion of intervals at all shouldn't be done. We're going to split things up so each interval gets its own json object. That means this structure is just going to be removed.